packages/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -168,12 +168,24 @@ static void InterpretInfoWindow(id<FLTGoogleMapMarkerOptionsSink> sink, NSDictio
 
 static UIImage* ExtractIcon(NSObject<FlutterPluginRegistrar>* registrar, NSArray* iconData) {
   UIImage* image;
+  static NSMutableDictionary* iconsCache;
+
+  if (iconsCache == nil) {
+    iconsCache = [NSMutableDictionary dictionary];
```

Comment:
May be this cache should have some configuration? User can have a lot of icons with different `hue` values. I think you should limit cache size